kernel/src/platform/system_timer.rs

```diff
@@ -0,0 +1,139 @@
+//! Interface system tick timer.
+
+use crate::hil::time::{self, Frequency};
+
+/// Interface for the system tick timer.
+///
+/// A system tick timer provides a countdown timer to enforce process scheduling
+/// quantums.  Implementations should have consistent timing while the CPU is
+/// active, but need not operate during sleep.
+///
+/// On most chips, this will be implemented by the core (e.g. the ARM core systick), but
+/// some chips lack this optional peripheral, in which case it might be
+/// implemented by another timer or alarm controller, or require virtualization
+/// on top of a single hardware timer.
+pub trait SystemTimer {
+    /// Sets the timer as close as possible to the given interval in
+    /// microseconds, and starts counting down. Interrupts are not enabled.
+    ///
+    /// Callers can assume at least a 24-bit wide clock. Specific timing is
+    /// dependent on the driving clock. In practice, increments of 10ms are most
+    /// accurate and values up to 400ms are valid.
```

Comment:
Are these comments universally true, or a holdover of the ARM implementation?

Reply:
These are a holdover from the ARM implementation. It might be reasonable to add a function to the trait to provide the minimum valid interval. I think it makes sense to leave the maximum interval as 400 ms -- RISC-V has no limitations on the maximum systick, but I think keeping schedulers platform independent is preferable and I don't imagine many use cases requiring timeslices that long.

Reply:
I update the comment.